test/src/e2e_vm_tests/mod.rs

```diff
@@ -1,730 +1,349 @@
-                "should_pass/require_contract_deployment/call_increment_contract",
-            ),
-            1, // true
-        ),
-        (
-            (
-                "should_pass/test_contracts/auth_testing_contract",
-                "should_pass/require_contract_deployment/caller_auth_test",
-            ),
-            1, // true
-        ),
-        (
-            (
-                "should_pass/test_contracts/context_testing_contract",
-                "should_pass/require_contract_deployment/caller_context_test",
-            ),
-            1, // true
-        ),
-        (
-            (
-                "should_pass/test_contracts/balance_test_contract",
-                "should_pass/require_contract_deployment/bal_opcode",
-            ),
-            1, // true
-        ),
-        (
-            (
-                "should_pass/test_contracts/test_fuel_coin_contract",
-                "should_pass/require_contract_deployment/token_ops_test",
-            ),
-            1, // true
-        ),
-        (
-            (
-                "should_pass/test_contracts/storage_access_contract",
-                "should_pass/require_contract_deployment/storage_access_caller",
-            ),
-            1, // true
-        ),
-        (
-            (
-                "should_pass/test_contracts/get_storage_key_contract",
-                "should_pass/require_contract_deployment/get_storage_key_caller",
-            ),
-            1,
-        ),
-        (
-            (
-                "should_pass/test_contracts/array_of_structs_contract",
-                "should_pass/require_contract_deployment/array_of_structs_caller",
-            ),
-            1,
-        ),
-        (
-            (
-                "should_pass/test_contracts/abi_with_tuples_contract",
-                "should_pass/require_contract_deployment/call_abi_with_tuples",
-            ),
-            1,
-        ),
-    ];
-
-    let total_number_of_tests = positive_project_names_no_abi.len()
-        + positive_project_names_with_abi.len()
-        + negative_project_names.len()
-        + contract_and_project_names.len();
-
-    // Filter them first.
-    let (contracts_and_projects, vals): (Vec<_>, Vec<_>) = contract_and_project_names
-        .iter()
-        .filter(|names| filter(names.0 .1))
-        .cloned()
-        .unzip();
-
-    let (contracts, projects): (Vec<_>, Vec<_>) = contracts_and_projects.iter().cloned().unzip();
-
-    // Deploy and then test.
-    number_of_tests_run += projects.len();
-    let mut contract_ids = Vec::<fuel_tx::ContractId>::with_capacity(contracts.len());
-    for name in contracts {
-        let contract_id = harness::deploy_contract(name, locked);
-        contract_ids.push(contract_id);
-    }
+            TestCategory::Compiles => {
+                assert!(crate::e2e_vm_tests::harness::compile_to_bytes(&name, locked).is_ok());
+                number_of_tests_run += 1;
```

Comment:
Should we be counting tests that don't run as ones that do run? Compilation, skipping etc. should not be counted as being run, no? Am I misunderstanding?

Reply:
We currently count all tests that are 'tested'.  So maybe that variable should just be `number_of_tests_executed`.  Not that important, could change it one day.

Reply:
I've already had to do a rebase to take in @vaivaswatha's PR changes, so I renamed this var while I was at it.